src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/HandlerResolver.cpp

```diff
@@ -214,14 +241,25 @@ try
     DWORD          dwBufferSize = s_initialGetNativeSearchDirectoriesBufferSize;
     DWORD          dwRequiredBufferSize = 0;
 
-    m_hHostFxrDll.Load(hostfxrOptions.GetHostFxrLocation());
-
+    try
+    {
+        m_hHostFxrDll.Load(hostfxrOptions.GetHostFxrLocation());
+    }
+    catch (...)
+    {
+        errorContext.detailedErrorContent = "Could not load hostfxr.dll.";
+        errorContext.statusCode = 500i16;
+        errorContext.subStatusCode = 32i16;
+        errorContext.generalErrorType = "ANCM Failed to Load dll";
+        errorContext.solution = "Most likely, the application was published for a different bitness than w3wp.exe/iisexpress.exe is running as.";
```

Comment:
Is it possible to know for sure what went wrong?

```suggestion
        errorContext.solution = "The application was likely published for a different bitness than w3wp.exe/iisexpress.exe is running as.";
```